### PR TITLE
Issue 296 allow complex paths

### PR DESCRIPTION
**Pull request by ChristianLangmann**
Allow complex paths in mulefd-components using xpath-expressions. 

Not really finished yet, as the displayed path is the XPath, and not descriptive

### CODE REVIEW

File: src/main/java/com/javastreets/mulefd/MuleXmlElement.java [hunk 1]
```diff
@@ -102,47 +110,60 @@ public static List<MuleComponent> fillComponents(Element flowElement,
...
+    return Optional.empty();
+  }
+
+  private static boolean isXpath(final String expression) {
+    return expression.startsWith(XPATH_INDICATOR);
+  }
 
+  private static String evaluateXpathOnElement(final Element element, final String expression) throws XPathExpressionException {
+    XPath xPath = XPathFactory.newInstance().newXPath();
+    return (String)xPath.compile(expression.substring(XPATH_INDICATOR.length())).evaluate(element, XPathConstants.STRING);
```
Comment (manikmagar) [inline src/main/java/com/javastreets/mulefd/MuleXmlElement.java:166]: Does `XPathExpressionException` must bubble up? Can it be wrapped and thrown as an instance of `com.javastreets.mulefd.drawings.DrawingException` (unchecked exception)?

**Comment by manikmagar** [comment]
Hi @ChristianLangmann, wanted to give an heads up about upcoming code refactor for #291 (with 0.11.0). Would love to get this xpath support for upcoming [0.10.0](https://github.com/manikmagar/mulefd/milestone/1) version before refactoring. Please feel free to let me know for anything on this one. I am happy to collaborate on this PR and make changes if you are okay with that. TIA!

File: src/main/java/com/javastreets/mulefd/MuleXmlElement.java (hunk 1 above)
Comment (ChristianLangmann) [reply src/main/java/com/javastreets/mulefd/MuleXmlElement.java:166]: Sure, can be done this way. Is probably nicer as it does not spoil the code as much.

**Comment by ChristianLangmann** [comment]
Hi, thanks for letting me know! Sure, like the idea of refactoring and finishing xpath-support before. I can clean up my fork for the pull-request if that helps?! Anything else I could do?

**Comment by manikmagar** [comment]
Hi! Yes, if you could do that, it would help. The latest code has a new `MuleFDException` (unchecked) that is used to throw any errors during component processing. That should avoid changing other method signatures for a checked exception. Thank you!

**Comment by manikmagar** [comment]
I couldn't make changes to this PR due to permissions, I guess. I have staged cleaned up code to separate branch, diff can be seen [here](https://github.com/manikmagar/mulefd/compare/feat/allow-complex-path?expand=1). Feel free to merge that into this PR if you okay. Or could just create a PR from there. I may release 0.10.0 tomorrow. Thank you!

File: src/main/java/com/javastreets/mulefd/MuleXmlElement.java (hunk 1 above)
Comment (manikmagar) [reply src/main/java/com/javastreets/mulefd/MuleXmlElement.java]: Resolved in https://github.com/manikmagar/mulefd/pull/300/commits/96008f40d4cd1db59539a3c7a9bd7fc529314f8d

**Comment by manikmagar** [comment]
I pushed commits to base PR branch to merge upstream and simplify the xpath processing. The core xpath processing logic remains the same as the original. PR is ready to review now.

**Pull request merged**